src/p_interaction.cpp

```diff
@@ -112,28 +112,32 @@ void P_TouchSpecialThing (AActor *special, AActor *toucher)
 
 
 // [RH]
-// SexMessage: Replace parts of strings with gender-specific pronouns
+// ProMessage: Replace parts of strings with player-specific pronouns
 //
 // The following expansions are performed:
-//		%g -> he/she/it
-//		%h -> him/her/it
-//		%p -> his/her/its
+//		%g -> he/she/they/it
+//		%h -> him/her/them/it
+//		%p -> his/her/their/its
+//		%s -> his/hers/theirs/its
+//		%r -> he's/she's/they're/it's
 //		%o -> other (victim)
 //		%k -> killer
 //
-void SexMessage (const char *from, char *to, int gender, const char *victim, const char *killer)
+void ProMessage (const char *from, char *to, int pronoun, const char *victim, const char *killer)
 {
-	static const char *genderstuff[3][3] =
+	static const char *pronouns[GENDER_MAX][5] =
 	{
-		{ "he",  "him", "his" },
-		{ "she", "her", "her" },
-		{ "it",  "it",  "its" }
+		{ "he",   "him",  "his",   "his",    "he's"    },
+		{ "she",  "her",  "her",   "hers",   "she's"   },
+		{ "they", "them", "their", "theirs", "they're" },
```

Comment:
Biggest issue I had with this is that this is a plural pronoun substitution to refer to a single entity. I know it's less offensive than "it" but it's also grammatically incorrect.

Reply:
There is a [long and well-documented history of singular they](https://en.wikipedia.org/wiki/Singular_they), which is not grammatically incorrect in the slightest. "Themself" is not standard English yet, but I have confidence that it will be in the future and so a reflexive pronoun type has not been added. This also improves backwards compatibility.

Reply:
"they" as a singular pronoun has been in use since the 14th century and is actually considered perfectly grammatical by multiple English dictionaries and style guides (including Merriam-Webster, the AP Stylebook, and Chicago Manual of Style).